### PR TITLE
Keyboard search on focused select without opening the overlay

### LINKED ISSUE #184: Focused select should also allow selecting an item with the keyboard without opening the overlay
Focused select should also allow selecting an item with the keyboard without opening the overlay. I'd expect select to behave like native HTML select.

### PR DESCRIPTION
**Pull request by SpaVec**
Fixes #184 
Added functionality to search and cycle through items with the keyboard when the select is focused, without opening the overlay.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-select/185)

### CODE REVIEW

**Comment by tomivirkki** [comment]
Thanks for the contribution @SpaVec. If we only changed the behaviour in `vaadin-select`, it would behave differently when the select is closed and when it's opened, which is unexpected. So, the logic should go to `vaadin-list-mixin` (where the current logic for keyboard selection is): https://github.com/vaadin/vaadin-list-mixin/blob/master/vaadin-list-mixin.html#L148

..the new logic should preferably be in a separate (static) function that could then be invoked from `vaadin-select`'s `_onKeyDown` while the select is still closed.

Please also include tests for this new feature. Generally all new code should be properly tested https://github.com/vaadin/vaadin/blob/master/DEVELOPMENT.md#running-tests

**Comment by SpaVec** [comment]
Good point, I'll do that.

Should the new function necessarily be static ? It seems easier to make it a method since I can make use of the _getAvailableIndex method.

I have created a pull request for vaadin-list-mixin here : https://github.com/vaadin/vaadin-list-mixin/pull/50

Waiting for your review on the list mixin to commit the changes for the select.

File: bower.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "vaadin-themable-mixin": "vaadin/vaadin-themable-mixin#^1.3.2",
     "vaadin-text-field": "vaadin/vaadin-text-field#^2.1.1",
     "vaadin-list-box": "vaadin/vaadin-list-box#^1.1.0",
+    "vaadin-list-mixin": "vaadin/vaadin-list-mixin#^2.2.0",
```
Comment (web-padawan) [inline bower.json]: This change doesn't seem right to me. The list-mixin should be picked up as a transitive dependency of the list-box. So we should release 1.2.0 of the list-box first.

Comment (web-padawan) [reply bower.json]: Discussed in person with @tomivirkki and I agree with his reasoning stated above.

**Pull request merged**